apps/api/lib/view/create.js

```diff
@@ -0,0 +1,31 @@
+import { schema } from '@tpluscode/rdf-ns-builders'
+import { CONSTRUCT } from '@tpluscode/sparql-builder'
+import fromStream from 'rdf-dataset-ext/fromStream.js'
+
+/**
+ * Queries the metadata endpoint to load necessary metadata for a view
+ *
+ * @param req
+ * @param pointer The payload, which is new instance of `</api/View>`
+ */
+export async function importMetadata({ req, pointer }) {
+  const sourceDataset = pointer.out(schema.isBasedOn).term
+  const view = pointer.term
+
+  if (sourceDataset) {
+    const metadata = await CONSTRUCT`
+      ${view} ${schema.identifier} ?identifier ;
+              ${schema.alternateName} ?alternateName ;
+              ${schema.name} ?name .
```

Comment:
It is necessary to add these to the view. Otherwise, SHACL validation would fail